test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -200,6 +201,37 @@ public void execute_list_showsAllPersons() throws Exception {
                               true,
                               expectedList);
     }
+    
+    @Test
+    public void execute_list_showsAllTags() throws Exception {
+        // prepare expectations
+        TestDataHelper helper = new TestDataHelper();
+        AddressBook expectedAB = helper.generateAddressBook(false, true);
+        List<? extends ReadOnlyTag> expectedList = expectedAB.getAllTags().immutableListView();
+
+        // prepare address book state
+        helper.addToAddressBook(addressBook, false, true);
+
+        // Execute the command
+        CommandResult r = logic.execute("listtag");
+
+        // Confirm the result contains the right data
+        assertEquals(Command.getMessageForTagListShownSummary(expectedList), r.feedbackToUser);
+        assertEquals(r.getRelevantTags().isPresent(), true);
+        if (true) {
+            assertEquals(expectedList, r.getRelevantTags().get());
+        }
+
+        // Confirm the state of data is as expected
+        assertEquals(expectedAB, addressBook);
+        assertEquals(expectedList, logic.getLastShownTagList());
+        assertEquals(addressBook, saveFile.load());
+    }
+    
+    @Test
+    public void hello() {
```

Comment:
@jaeoheeail It test if the listtag command can correctly list out all the tags in the address book.